### PR TITLE
Migrate translator/internaldata 

### PR DESCRIPTION
**Pull request by alexperez52**
**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/3474

**Description:** <Describe what has changed.>
 This PR migrates `translator/internaldata` to `pkg/translator/opencensus/`

### CODE REVIEW

**Comment by alexperez52** [comment]
This is missing import formatting

**Review by bogdandrutu — COMMENTED** [review]
The package name should be "opencensus"

File: pkg/translator/opencensus/internaldata/metrics_to_oc.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package internaldata
+package opencensus
```
Comment (Aneurysm9) [inline pkg/translator/opencensus/internaldata/metrics_to_oc.go:15]: should these files all be up one level?

Comment (alexperez52) [reply pkg/translator/opencensus/internaldata/metrics_to_oc.go:15]: Outside of the internaldata folder ?

Comment (Aneurysm9) [reply pkg/translator/opencensus/internaldata/metrics_to_oc.go:15]: yes, in `pkg/translator/opencensus`.  Would align the package name with the import path and avoid an extra element in the import path that adds no value.

File: pkg/translator/opencensus/internaldata/metrics_to_oc_test.go [hunk 2]
```diff
@@ -12,23 +12,24 @@
...
-package internaldata
+package opencensus
 
 import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+
```
Comment (bogdandrutu) [inline pkg/translator/opencensus/internaldata/metrics_to_oc_test.go]: Extra whitespace here?

File: pkg/translator/opencensus/internaldata/metrics_to_oc.go (hunk 1 above)
Comment (alexperez52) [reply pkg/translator/opencensus/internaldata/metrics_to_oc.go]: I have just pushed a commit that addresses this

File: pkg/translator/opencensus/traces_to_oc.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	"go.opencensus.io/trace"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
-	"go.opentelemetry.io/collector/internal/occonventions"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/occonventions"
```
Comment (bogdandrutu) [inline pkg/translator/opencensus/traces_to_oc.go:27]: I don't see a dependency on this

Comment (alexperez52) [reply pkg/translator/opencensus/traces_to_oc.go:27]: line 195 uses `val, ok := attr.Get(occonventions.AttributeSameProcessAsParentSpan)`
should it be removed from imports ?

Comment (bogdandrutu) [reply pkg/translator/opencensus/traces_to_oc.go:27]: In the go mod

Comment (alexperez52) [reply pkg/translator/opencensus/traces_to_oc.go]: I see. Ive added it

File: pkg/translator/opencensus/go.mod [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus
+
+go 1.16
+
+require (
+	github.com/census-instrumentation/opencensus-proto v0.3.0
+	github.com/golang/protobuf v1.5.2
+	github.com/google/go-cmp v0.5.6
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-20210820201140-13cd75901f9f
```
Comment (bogdandrutu) [inline pkg/translator/opencensus/go.mod:9]: Here overwrite with `	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-00010101000000-000000000000
`

then at the bottom of the file add:

replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal => ../../internal/coreinternal

Comment (alexperez52) [reply pkg/translator/opencensus/go.mod]: Done

**Comment by Aneurysm9** [comment]
Check the relative path to this test data file:

```
--- FAIL: TestInternalTracesToOCTracesAndBack (0.00s)
    traces_to_oc_test.go:361: 
        	Error Trace:	traces_to_oc_test.go:361
        	Error:      	Received unexpected error:
        	            	open ../../internal/goldendataset/testdata/generated_pict_pairs_traces.txt: no such file or directory
        	Test:       	TestInternalTracesToOCTracesAndBack
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus	0.066s
```

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**